data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/sink/KafkaSink.java

```diff
@@ -0,0 +1,166 @@
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSinkConfig;
+import org.opensearch.dataprepper.plugins.kafka.producer.KafkaSinkProducer;
+import org.opensearch.dataprepper.plugins.kafka.producer.ProducerWorker;
+import org.opensearch.dataprepper.plugins.kafka.util.SinkPropertyConfigurer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation class of kafka--sink plugin. It is responsible for receive the collection of
+ * {@link Event} and produce it to different kafka topics.
+ */
+@DataPrepperPlugin(name = "kafka", pluginType = Sink.class, pluginConfigurationType = KafkaSinkConfig.class)
+public class KafkaSink extends AbstractSink<Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaSink.class);
+
+    private final KafkaSinkConfig kafkaSinkConfig;
+
+    private volatile boolean sinkInitialized;
+
+    private static final Integer totalWorkers = 1;
+
+    private ProducerWorker producerWorker;
+
+    private ExecutorService executorService;
+
+    private final PluginFactory pluginFactory;
+
+    private final PluginSetting pluginSetting;
+
+    private final ExpressionEvaluator expressionEvaluator;
+
+    private final SinkContext sinkContext;
+
+
+    @DataPrepperPluginConstructor
+    public KafkaSink(final PluginSetting pluginSetting, final KafkaSinkConfig kafkaSinkConfig, final PluginFactory pluginFactory,
+                     final ExpressionEvaluator expressionEvaluator, final SinkContext sinkContext) {
+        super(pluginSetting);
+        this.pluginSetting = pluginSetting;
+        this.kafkaSinkConfig = kafkaSinkConfig;
+        this.pluginFactory = pluginFactory;
+        this.expressionEvaluator = expressionEvaluator;
+        this.sinkContext = sinkContext;
+
+
+    }
+
+
+    @Override
+    public void doInitialize() {
+        try {
+            doInitializeInternal();
+        } catch (InvalidPluginConfigurationException e) {
+            LOG.error("Invalid plugin configuration, Hence failed to initialize kafka-sink plugin.");
+            this.shutdown();
+            throw e;
+        } catch (Exception e) {
+            LOG.error("Failed to initialize kafka-sink plugin.");
+            this.shutdown();
+            throw e;
+        }
+    }
+
+    private void doInitializeInternal() {
+        executorService = Executors.newFixedThreadPool(totalWorkers);
+        sinkInitialized = Boolean.TRUE;
+    }
+
+    @Override
+    public void doOutput(Collection<Record<Event>> records) {
+        if (records.isEmpty()) {
+            return;
+        }
+        try {
+            final KafkaSinkProducer producer = createProducer();
+            records.forEach(record -> {
```

Comment:
For each record do we need to create Producer worker.? Please check how Consumer thread is created,

Reply:
yes, this is as per the requirement only.